RELEASE.md

```diff
@@ -0,0 +1,106 @@
+# Release process
+
+This file documents the necessary steps for releasing Irmin to its various users
+(via GitHub, `opam-repository` and Tezos).
+
+At a high level, releasing Irmin consists of publishing the following artefacts:
+
+- a Git [commit tag][git-tags];
+- a set of documentation on GitHub pages (e.g. [`mirage.github.io/irmin`][pages-docs]);
+- a release archive (`.tbz` file containing the project source) on GitHub;
+- a set of `.opam` files in [`opam-repository`][opam-repo] that point to this
+  release archive;
+- (optionally) a copy of these `.opam` files in the Tezos
+  [`opam-repository`][tezos-opam-repo].
+
+Most of this can be handled automatically by [`dune-release`][dune-release], as
+described in the instructions below.
+
+[git-tags]: https://git-scm.com/book/en/v2/Git-Basics-Tagging
+[pages-docs]: https://mirage.github.io/irmin
+[dune-release]: https://github.com/ocamllabs/dune-release
+[opam-repo]: https://github.com/ocaml/opam-repository
+
+## Releasing to opam-repository and GitHub
+
+- Check that no `.opam` files contain `pin-depends` fields. If so, release those
+  packages first.
+
+```sh
+; git grep -A 10 "pin-depends" *.opam
```

Comment:
Can you open an issue on dune-release to track this with `dune-release lint` ? /cc @pitag-ha

Reply:
That's a good point, thanks! I've just opened [that issue](https://github.com/ocamllabs/dune-release/issues/356) in dune-release.

So from what I understand the `dune-release` lints could catch if one has forgotten to remove the `pin-depends` field before releasing. This step @CraigFe is doing here of finding out which packages are in `pin-depends` (if any) and releasing them first, would still have to be performed manually though. Is that right?

Reply:
Yep, it'd still be manual. I suppose the corresponding feature here is a multi-repository release process, but that sounds like a big feature.